src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java/JavaLanguageLexer.g4

```diff
@@ -334,6 +334,39 @@ STRING_TEMPLATE_END:     {stringTemplateDepth > 0}?
                          {stringTemplateDepth--;}
                          ;
 
+// Text block fragments
+
+fragment TextBlockContent
+    : ( TwoDoubleQuotes
+      | OneDoubleQuote
+      | Newline
+      | TextBlockCharacter
+      )+
+    ;
+
+fragment TextBlockCharacter
+    : ~["\\]
+    | TextBlockStandardEscape
+    | EscapeSequence
+    ;
+
+fragment TextBlockStandardEscape
+    : '\\' ( [btnfrs'\\] | Newline | OneDoubleQuote )
```

Comment:
this is not a minor change, please use `Pull #xxxxx` to keep strong reference to reasons/comments why we did this update.

Reply:
Done